### DISCUSSION TITLE
Semester 4 Week 6

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#219] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.
Again a week has approached and we have a few update for you:

### Frontend

✅ Implemented login functionality 
- When the user gets logged in a token will be created and a session get started until log out

✅  Fixed misplaced speech bubble of Carlos in the login- and landing page

✅  Implemented a Help button and SpeechBubble in Food Forge

✅ Replaced static logo in the NavBar with a funky .gif

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/e409ed31-d77b-4029-bd04-7b557c6fe759" height="200" />

✅ Redesigned NavBar because CookPit didn't like it.

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/cd2f208b-bf4a-41ca-96bd-b1cd33e4768c" />

<br>
<br>

## Testing

✅ LEN wrote thousands of millions of API tests with xUnit.NET 

![image](https://github.com/SE-TINF22B6/CookHub/assets/113697513/6e442c7c-b7e9-44bc-9ab9-cb053caabeae)
...

<br>
<br>

## Backend

✅ Adventurized texts are now saveable in the database

✅ Created classes for database entities which only include attributes relevant for the frontend
- Requests only returns objects of these classes
- thats ensures no sensitive information gets leaked

✅ Fixed bug preventing creation of recipes

<br><br>

## Metrics

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/7dd873f2-e135-44c1-9b84-a21aed299dcb" height="150" />

<br>

Our [Web Application Monitoring Tools Presentation](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Presentation/Web%20application%20monitoring%20tools.pdf)

### Our metrics which will be measured for our project:

- **Google Lighthouse** -> measuring response time  and other performance/quality metrics of website

- **DotCover** (integrated in the Rider IDE) -> measuring est coverage

![image](https://github.com/SE-TINF22B6/CookHub/assets/113697513/40d12b18-d382-4cdf-9e11-81fb0e35dc39)

- **NDepend** -> measuring code quality

![image](https://github.com/SE-TINF22B6/CookHub/assets/77683850/290d3e21-d300-4370-95c4-4890c43419e9)

## Team Building

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/5d6031d5-3737-4c44-b328-838235d223f0" height="200" /> <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/a679ee56-c0e5-4fac-b573-ebbb32464960" height="200" /> <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/ffb2583c-01f6-497e-ac0d-fe6145a17a0a" height="200" />

<br><br>

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by giomskii** [comment]
Hello Team CookHub,
its nice to see that you have fully implemented your login process. The new gif in the navbar also looks better than the static version.
As always, team building is very important 👍🏼 
Keep up with the good work! - Team DHBWhub

**Comment by BenjaminB109** [comment]
Dear CookHub Team,
I'm glad to hear that you were able to implement the login functionality. The funky GIF looks fabulous.

**Comment by fehlend0** [comment]
Hello CookHub Team, 

it's great to see the implementation of login functionality. The addition of a Help button and SpeechBubble in Food Forge is a thoughtful touch. And that .gif logo? Brilliant! 

Looking forward to the next update, 
Your Statify Team

**Comment by NicolasVien** [comment]
I'm really impressed with the amount of work you've done and the progress you've made on the project. Your work on the frontend is exceptional, and the login implementation is top-notch. It's also great to see the progress you've made on the database, and your metrics seem well-suited for your web application. Keep up the excellent work, and I have no doubt that you'll continue to make great strides on the project.
